xbmc/filesystem/CurlFile.cpp

```diff
@@ -626,6 +626,10 @@ void CCurlFile::SetCommonOptions(CReadState* state, bool failOnError /* = true *
   // Setup allowed TLS/SSL ciphers. New versions of cURL may deprecate things that are still in use.
   if (!m_cipherlist.empty())
     g_curlInterface.easy_setopt(h, CURLOPT_SSL_CIPHER_LIST, m_cipherlist.c_str());
+
+  // enable HTTP2 support. default: CURL_HTTP_VERSION_1_1. Curl >= 7.62.0 defaults to CURL_HTTP_VERSION_2TLS
+  g_curlInterface.easy_setopt(h, CURLOPT_HTTP_VERSION, CURL_HTTP_VERSION_2_0);
```

Comment:
No support for building with older curl?

What is the problem with `CURL_HTTP_VERSION_2TLS`?

Reply:
`CURL_HTTP_VERSION_2_0` was added in version 7.33.0 (14th October 2013). So why would you want to support even older versions?

Reply:
> CURL_HTTP_VERSION_2_0 was added in version 7.33.0 (14th October 2013). So why would you want to support even older versions?

Crazy how time flies. To me HTTP2 still seemed to be a totally recent technology :-) Disregard then.

Reply:
@Rechi could you please test if windows is ok with this without nghttp2 or if I need to make this posix only?

Reply:
updating & installing addons and playing videos from http & https still works